assertj-core/src/main/java/org/assertj/core/presentation/StandardRepresentation.java

```diff
@@ -401,6 +402,10 @@ protected String toStringOf(Class<?> c) {
   }
 
   protected String toStringOf(String s) {
+    return toStringOf((CharSequence) s);
+  }
+
+  protected String toStringOf(CharSequence s) {
     return concat("\"", s, "\"");
   }
```

Comment:
I wasn't sure whether the existing `toStringOf(String)` should be replaced (since `String` implements `CharSequence`), or whether that would be a breaking API change.

Reply:
It's not a binary breaking change but if some users rely on that representation then it would be, so I'm inclined we don't change it. Reusing the String representation for char sequence is ok as if the contents are the same we would add the type to differentiate the values.

Reply:
Ah sorry my comment was a bit misleading. What I meant is:
- Should there be `toStringOf(String)` and `toStringOf(CharSequence)` (as done here currently)
- Or, should there only be `toStringOf(CharSequence)`, and `toStringOf(String)` is removed\
The code would then simply use the `CharSequence` variant for `String` as well. But removing the `toStringOf(String)` overload would technically be an incompatible API change since it is `protected` and users might have overridden it.

Reply:
I think it would be better to extract `concat("\"", s, "\"");` to a private method and call it from both `toStringOf`, users can then override `toStringOf(CharSequence s)` without impacting `toStringOf(String s)`.

Reply:
I hope the latest changes in 6952df829991457aed6cd10b667e8e27b68e56af are what you had in mind.

Reply:
thanks, I'll review that next week, I'm off for a few days